tests/unit/testsuite/Bolt/Boltpay/Helper/UrlTraitTest.php

```diff
@@ -49,110 +49,74 @@ protected function tearDown()
 
     /**
      * @test
-     * that getBoltMerchantUrl returns production merchant url when sandbox mode is set to false in configuration
-     *
-     * @covers Bolt_Boltpay_Helper_UrlTrait::getBoltMerchantUrl
-     *
-     * @throws ReflectionException if trait doesn't have merchantUrlProd property
-     * @throws Mage_Core_Model_Store_Exception from test helper if store doesn't exist
+     * @covers Bolt_Boltpay_Helper_UrlTrait::getApiUrl
+     * @dataProvider providerGetApiUrl
      */
-    public function getBoltMerchantUrl_withTestConfigurationSetToFalse_returnsProductionMerchantUrl()
-    {
-        Bolt_Boltpay_TestHelper::stubConfigValue('payment/boltpay/test', false);
+    public function getApiUrl($test, $customValue, $result) {
+        Bolt_Boltpay_TestHelper::stubConfigValue('payment/boltpay/test', $test);
+        Bolt_Boltpay_TestHelper::stubConfigValue('payment/boltpay/custom_api', $customValue);
         $this->assertEquals(
-            Bolt_Boltpay_TestHelper::getNonPublicProperty($this->currentMock, 'merchantUrlProd'),
-            $this->currentMock->getBoltMerchantUrl()
+            $result,
+            $this->currentMock->getApiUrl()
         );
     }
 
-    /**
-     * @test
-     * that getBoltMerchantUrl returns sandbox merchant url when sandbox mode is set to true in configuration
-     *
-     * @covers Bolt_Boltpay_Helper_UrlTrait::getBoltMerchantUrl
-     *
-     * @throws ReflectionException if trait doesn't have merchantUrlSandbox property
-     * @throws Mage_Core_Model_Store_Exception from test helper if store doesn't exist
-     */
-    public function getBoltMerchantUrl_withTestConfigurationSetToTrue_returnsSandboxMerchantUrl()
-    {
-        Bolt_Boltpay_TestHelper::stubConfigValue('payment/boltpay/test', true);
-        $this->assertEquals(
-            Bolt_Boltpay_TestHelper::getNonPublicProperty($this->currentMock, 'merchantUrlSandbox'),
-            $this->currentMock->getBoltMerchantUrl()
-        );
+    public function providerGetApiUrl() {
+        return [
+            [false, '', 'https://api.bolt.com/'],
+            [false, 'https://api.vitaliy.dev.bolt.me/', 'https://api.bolt.com/'],
+            [true, '', 'https://api-sandbox.bolt.com/'],
+            [true, 'https://wrong.url.com/', 'https://api-sandbox.bolt.com/'],
+            [true, 'https://api.vitaliy.dev.bolt.me/', 'https://api.vitaliy.dev.bolt.me/'],
```

Comment:
Could you specifically add a test for staging

Reply:
Good catch, added